x/evm/types/types.go

```diff
@@ -1212,6 +1269,9 @@ func DecodeTransferSinglesigParams(bz []byte) (common.Address, error) {
 	if err != nil {
 		return common.Address{}, err
 	}
+	if actual, err := arguments.Pack(params...); err != nil || !bytes.Equal(actual, bz) {
```

Comment:
Maybe a helper function for this? Seems like a recurrent piece of code across the Params functions